src/object/srv_obj.c

```diff
@@ -1135,16 +1175,32 @@ obj_local_rw(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 			goto out;
 		}
 	} else {
-		uint32_t	fetch_flags;
+		uint64_t			 cond_flags;
+		uint32_t			 fetch_flags;
+		bool				 ec_deg_fetch;
+		struct daos_recx_ep_list	*shadows = NULL;
 
 		size_fetch = (!rma && orw->orw_sgls.ca_arrays == NULL);
 		fetch_flags = size_fetch ? VOS_FETCH_SIZE_ONLY : 0;
+		cond_flags = orw->orw_api_flags | VOS_OF_USE_TIMESTAMPS;
 		bulk_op = CRT_BULK_PUT;
 
+		ec_deg_fetch = orw->orw_flags & ORF_EC_DEGRADED;
+		if (ec_deg_fetch && !size_fetch) {
+			rc = obj_fetch_shadow(cont->sc_hdl, orw->orw_oid,
+				orw->orw_epoch, cond_flags, dkey, orw->orw_nr,
+				iods, orw->orw_tgt_idx, &shadows);
+			if (rc) {
+				D_ERROR(DF_UOID" Fetch shadow failed: "DF_RC
+					"\n", DP_UOID(orw->orw_oid), DP_RC(rc));
+				goto out;
+			}
+		}
+
 		rc = vos_fetch_begin(cont->sc_hdl, orw->orw_oid, orw->orw_epoch,
-				     orw->orw_api_flags | VOS_OF_USE_TIMESTAMPS,
-				     dkey, orw->orw_nr, iods, fetch_flags, NULL,
-				     &ioh, dth);
+				     cond_flags, dkey, orw->orw_nr, iods,
+				     fetch_flags, shadows, &ioh, dth);
+		daos_recx_ep_list_free(shadows, orw->orw_nr);
```

Comment:
style, easier to understand in this way, instead of checking with daos_recx_ep_list_free
```  
 if (shadow)
         daos_recx_ep_list_free
```

Reply:
ok